lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/passportCri/ContractTest.java

```diff
@@ -356,6 +509,154 @@ void fetchVerifiableCredential_whenCalledAgainstPassportCri_retrievesAValidVc(
                         });
     }
 
+    @Test
+    @PactTestFor(pactMethod = "validRequestReturnsIssuedCredentialWithCi")
+    void fetchVerifiableCredential_whenCalledAgainstPassportCri_retrievesAValidVcWithACi(
+            MockServer mockServer) throws URISyntaxException, CriApiException {
+        // Arrange
+        var credentialIssuerConfig = getMockCredentialIssuerConfig(mockServer);
+        ContraIndicatorConfig ciConfig1 = new ContraIndicatorConfig(null, 4, null, null);
+        ContraIndicatorConfig ciConfig2 = new ContraIndicatorConfig(null, 4, null, null);
+        Map<String, ContraIndicatorConfig> ciConfigMap = new HashMap<>();
+        ciConfigMap.put("D02", ciConfig1);
+
+        when(mockConfigService.getCriConfig(any())).thenReturn(credentialIssuerConfig);
+        when(mockConfigService.getCriPrivateApiKey(any())).thenReturn(PRIVATE_API_KEY);
+        // This mock doesn't get reached in error cases, but it would be messy to explicitly not set
+        // it
+        Mockito.lenient()
+                .when(mockConfigService.getContraIndicatorConfigMap())
+                .thenReturn(ciConfigMap);
+
+        var verifiableCredentialJwtValidator =
+                new VerifiableCredentialJwtValidator(
+                        mockConfigService,
+                        ((exactMatchClaims, requiredClaims) ->
+                                new FixedTimeJWTClaimsVerifier<>(
+                                        exactMatchClaims,
+                                        requiredClaims,
+                                        Date.from(CURRENT_TIME.instant()))));
+
+        // We need to generate a fixed request, so we set the secure token and expiry to constant
+        // values.
+        var underTest =
+                new CriApiService(
+                        mockConfigService, mockSigner, mockSecureTokenHelper, CURRENT_TIME);
+
+        // Act
+        var verifiableCredentialResponse =
+                underTest.fetchVerifiableCredential(
+                        new BearerAccessToken("dummyAccessToken"),
+                        getCallbackRequest("dummyAuthCode", credentialIssuerConfig),
+                        getCriOAuthSessionItem());
+
+        verifiableCredentialResponse
+                .getVerifiableCredentials()
+                .forEach(
+                        credential -> {
+                            try {
+                                verifiableCredentialJwtValidator.validate(
+                                        credential, credentialIssuerConfig, TEST_USER);
+
+                                JsonNode credentialSubject =
```

Comment:
This test is about CIs so I think it's worth checking that we can parse the CI from the response.